packages/drivers/routerlicious-driver/src/documentServiceFactory.ts

```diff
@@ -64,14 +64,7 @@ export class RouterliciousDocumentServiceFactory implements IDocumentServiceFact
         if (!parsedUrl.pathname) {
             throw new Error("Parsed url should contain tenant and doc Id!!");
         }
-        // extract tenant and document IDs from the parsed URL.
-        const [, tenantId, newDocumentId] = parsedUrl.pathname.split("/");
-        // TODO: This logic to determine ID helps with back-compat of the legacy container create flow.
-        // Will ignore the new document ID provided by the application as r11s will not honor it.
-        // The literal "new" document ID interpreted as a request to generate the ID
-        // by the server and therefore is replaced with `undefined` value.
-        // Any other value will be transmitted as-is to support the legacy container create flow.
-        const id = newDocumentId !== "new" ? newDocumentId : undefined;
+        const [, tenantId] = parsedUrl.pathname.split("/");
```

Comment:
Until the example apps like Clicker work with server-side document id, I'm not sure we should enforce it in the driver. What is the benefit of doing this client-side enforcing now?

Reply:
Needed to make sure tests are valid and consistent. Examples will be fixed shortly after.

Reply:
If this merges before the apps, then Clicker will become unusable until the app changes are in. Is it possible to merge the app changes first? We use Clicker very frequently for testing Azure Fluid Relay, so breaking the latest build with this change is not a good spot to be in.

Is my understanding correct that the tests will always pass in a documentId, so this change is required to make the driver ignore the test's generated document id and actually validate server-side ids?

Reply:
Agreed. Apps should go first still in the back-compat mode. Meaning, not passing the document ID but the driver remains compatible with tests.

Reply:
Regarding tests, they will keep sending the document ID for the back-compat validation, e.g. against 0.39 driver.

Reply:
Example apps are supported via the latest webpack loader merged earlier.